tests/api/onboarding-tasks.php

```diff
@@ -167,4 +174,353 @@ function ( $template ) {
+		TaskLists::add_list(
+			array(
+				'id' => 'test-list',
+			)
+		);
+
+		TaskLists::add_task(
+			'test-list',
+			array(
+				'id'            => 'test-task',
+				'title'         => 'Test Task',
+				'is_snoozeable' => true,
+			)
+		);
+
+		$request = new WP_REST_Request( 'POST', $this->endpoint . '/test-task/snooze' );
+		$request->set_headers( array( 'content-type' => 'application/json' ) );
+		$request->set_body( wp_json_encode( array( 'task_list_id' => 'test-list' ) ) );
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$task = TaskLists::get_task( 'test-task' );
+
+		$this->assertEquals( $data['isSnoozed'], true );
+		$this->assertEquals( isset( $data['snoozedUntil'] ), true );
+		$this->assertEquals( $task->is_snoozed(), true );
+		$this->assertEquals( isset( $task->snoozed_until ), true );
+
+	}
+
+	/**
+	 * Test that a task can be snoozed with determined duration.
+	 * @group tasklist
+	 */
+	public function test_task_can_be_snoozed_with_duration() {
+		wp_set_current_user( $this->user );
+
+		TaskLists::add_list(
+			array(
+				'id' => 'test-list',
+			)
+		);
+
+		TaskLists::add_task(
+			'test-list',
+			array(
+				'id'            => 'test-task',
+				'title'         => 'Test Task',
+				'is_snoozeable' => true,
+			)
+		);
+
+		$request = new WP_REST_Request( 'POST', $this->endpoint . '/test-task/snooze' );
+		$request->set_headers( array( 'content-type' => 'application/json' ) );
+		$request->set_body( wp_json_encode( array( 'duration' => 'week' ) ) );
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$task = TaskLists::get_task( 'test-task' );
+
+		$week_in_ms = WEEK_IN_SECONDS * 1000;
+
+		$this->assertEquals( $data['snoozedUntil'] >= ( ( time() * 1000 ) + $week_in_ms ), true );
+
+	}
+
+	/**
+	 * Test that a snoozed task can be undone.
+	 * @group tasklist
+	 */
+	public function test_snoozed_task_can_be_undone() {
+		wp_set_current_user( $this->user );
+
+		TaskLists::add_list(
+			array(
+				'id' => 'test-list',
+			)
+		);
+
+		TaskLists::add_task(
+			'test-list',
+			array(
+				'id'            => 'test-task',
+				'title'         => 'Test Task',
+				'is_snoozeable' => true,
+			)
+		);
+
+		$task = TaskLists::get_task( 'test-task' );
+
+		$task->snooze();
+
+		$this->assertEquals( $task->is_snoozed(), true );
+
+		$request = new WP_REST_Request( 'POST', $this->endpoint . '/test-task/undo_snooze' );
+		$request->set_headers( array( 'content-type' => 'application/json' ) );
+		$response = $this->server->dispatch( $request );
+		$data     = $response->get_data();
+
+		$task_after_request = TaskLists::get_task( 'test-task' );
+
+		$this->assertEquals( $task_after_request->is_snoozed(), false );
+
+	}
+
+	/**
+	 * Test that snooze endpoint returns error for invalid task.
+	 * @group tasklist
+	 */
+	public function test_snoozed_task_invalid() {
+		$this->markTestSkipped( 'Skipped temporarily due to change in endpoint behavior.' );
```

Comment:
Setting this test to be skipped for the moment to gather more info on expected behavior here.

Reply:
Nice!  Thanks for adding this 👍